Added GitHub workflow for AltStore Repo

sercansahan opened the pull request:
This addresses #102 and #484 

Basically, created a `apps.json` (needed for AltStore source, may need manual updating for screenshots since I used screenshots from [Aidoku Website](https://aidoku.app) ), `update_json.py` to read and update that json file with each GitHub Release and `.github/workflows/update_source.yml` to run a GitHub Workflow.

These three files are based on the same files in [YTLitePlus-Altstore](https://github.com/YTLitePlus/YTLitePlus-Altstore)

`apps.json` and `update_json.py` use `app.aidoku.Aidoku` as Bundle ID while latest release (v0.6.10) uses `xyz.skitty.Aidoku` as Bundle ID. As a result, latest release does not install via AltStore Source. After a new version with new Bundle ID is released, install via AltStore Source will work.

If Minimum Deployment Version is updated in the future, value for `minimum_ios_version` must be updated (8th line in `update_json.py`). 

URL to `apps.json`  file can be used with `altstore://source?url=<url_to_apps.json>` to enable adding source with one click if AltStore is installed.

Full documentation about `apps.json` can be found at [Make a Source | AltStore](https://faq.altstore.io/developers/make-a-source)

Skittyblock commented on .github/workflows/update_source.yml:
```diff
@@ -0,0 +1,74 @@
+        echo "" >> $GITHUB_STEP_SUMMARY
+        echo "📆 Next Scheduled Run" >> $GITHUB_STEP_SUMMARY
+        echo "" >> $GITHUB_STEP_SUMMARY
+        echo "The next scheduled run will be tomorrow at midnight UTC." >> $GITHUB_STEP_SUMMARY
```
should have a newline at the end of all files
```suggestion
        echo "The next scheduled run will be tomorrow at midnight UTC." >> $GITHUB_STEP_SUMMARY

```

Skittyblock reviewed (changes requested):
thanks, this looks like it might actually be helpful for people. but a few things:
- i don't really like the idea of having the `apps.json` on the main branch, with extra commits updating it. we should have it on a separate branch, like how the [workflow for the community sources](https://github.com/Skittyblock/aidoku-community-sources/blob/main/.github/workflows/build.yaml) does it
  - the branch should probably be called `altstore` and it should just have the single commit, the same as community sources
- the `update_json.py` is also only used for the workflow, so it should be in a "supporting" folder inside the worflows folder

sercansahan commented:
Ok, just did what you suggested

- Moved `update_json.py` to `.github/workflows/supporting`
- Updated `update_source.yml` file so it does not commit to main branch. Instead, it uses branch named `altstore`

Skittyblock reviewed (changes requested):
like the community sources workflow, i think the only thing we need in the altstore branch is the `apps.json` file, and should only have the single commit adding it. it shouldn't be based on main if possible. the python script will need to be updated.

sercansahan commented on .github/workflows/update_source.yml:
```diff
@@ -0,0 +1,100 @@
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "This job took ${{ steps.duration.outputs.duration }} to complete." >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "📆 Next Scheduled Run" >> $GITHUB_STEP_SUMMARY
```
Yes, there is no scheduled next run since this workflow is triggered by releases. These lines were left from original file. Removed them.

sercansahan commented:
Python script only gets releases from GitHub and updates local apps.json file, it does not need update apart from bundle id and minimum version changes. yml file is updated to match your request. Now, just like community sources repo, it uses github-pages-deploy-action to deploy apps.json on a separate branch. Only base json file stays in the main branch, the one to use as altstore source will be on altstore branch.

sercansahan commented:
That's great. Just remember, the action is triggered when a release is published. If you merge pull request after publishing a new release, you have to trigger the action manually.
